### PR TITLE
[12.0] [REF] Account Banking Pain Base: constraint on BIC

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**
Prevent inserting BIC not respecting the SEPA pain constraint.

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Comment by qgroulard** [comment]
A valid BIC is 8 or 11 characters long. 
It would be nice to check that too :)

File: account_banking_pain_base/models/res_bank.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @api.constrains('bic')
+    def _check_bic(self):
+        """
+        This method strengthens the constraint on the BIC of the bank account
+        (The account_payment_order module already checks the length in the
+         check_bic_length method).
+        :raise: ValidationError if the BIC doesn't respect the regex of the
+        SEPA pain schemas.
+        """
+        bic_re = r"[A-Z]{6,6}[A-Z2-9][A-NP-Z0-9]([A-Z0-9]{3,3}){0,1}"
```
Comment (sbidoul) [inline account_banking_pain_base/models/res_bank.py]: A few things, from important to nitpicking:
- add a `$` at the end, to make sure longer strings do not match
- use `re.compile` to precompile the regexp as a module-level constant
- `{0,1}` is more commonly written `?`
- `{6,6}` can also be written `{6}`

File: account_banking_pain_base/models/res_bank.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        invalid_banks = self.filtered(lambda r: not re.match(bic_re, r.bic))
+        if invalid_banks:
+            raise ValidationError(_(
+                "The following Bank Identifier Codes (BIC) does not respect "
```
Comment (sbidoul) [inline account_banking_pain_base/models/res_bank.py]: s/does/do/

**Review by sbidoul — CHANGES_REQUESTED** [review]
A few improvements

**Comment by ThomasBinsfeld** [comment]
@sbidoul Done

**Review by sbidoul — APPROVED** [review]

**Pull request merged**